dask/array/core.py

```diff
@@ -1376,7 +1376,7 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
         return self
 
     def __reduce__(self):
-        return (Array, (self.dask, self.name, self.chunks, self.dtype))
+        return (Array, (self.dask, self.name, self.chunks, self.dtype, self._meta))
```

Comment:
This makes me a little worried since `_meta` is a potentially large object. Would including `type(self._meta)` accomplish the same thing? You can see that's what the `layer.collection_annotations` does a few lines above.

Reply:
In this case the reduction function will transfer "_meta" to string. How do you want me to proceed to rebuild the instance from the string? 

I did not find in the code how you do that. A solution, would be to do something like this :

```python
def _from_string(x, *args, **kwargs):
    modulename, class_ = x.rsplit(".", 1)
    return getattr(importlib.import_module(modulename), class_)(*args, **kwargs)
```

But in this case, meta will lose information about the properties of the hidden array (fill_value).

Maybe you should pass "_meta" only in case it is an array or a masked array, because in these cases the objects will be small.

Reply:
@jsignell I tried this with @ncclementi as well. `Array` is expecting an empty ndarray, and `type(self._meta)` just returns the class name. When can we expect `_meta` to be large enough to cause issues here?

And, what do you think about @fbriol's suggestion above?

Reply:
I think the issue is that `_meta` is something that the user can set, so we can't really trust that it will be small. That being said, maybe serializing the whole `_meta` is fine. It is definitely a bad practice to have a big meta, so if we run into issues we can always be more restrictive about that. I hadn't thought about the properties being important @fbriol thanks for that example.